pkgs/by-name/go/gol/package.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, maven, fetchFromGitHub, makeWrapper, jre }:
+
+maven.buildMavenPackage rec {
+  pname = "gol";
+  version = "0.1.9";
+
+  src = fetchFromGitHub {
+    owner = "clarisma";
+    repo = "gol-tool";
+    rev = version;
+    hash = "sha256-UeuLXYGTmoAlrDm8c/Ibgi2q7G5MZJV3gGS+o5W3iOI=";
+  };
+
+  mvnHash = "sha256-GWGhx0pbojQhYshmrJ2sXMJJt4mhd3XBwAfzy9dCSMs=";
+  mvnParameters = "compile assembly:single -Dmaven.test.skip=true";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,lib}
+    cp /build/source/target/gol-tool-${version}-jar-with-dependencies.jar $out/lib/gol-tool.jar
+
+    makeWrapper ${jre}/bin/java $out/bin/gol \
+      --add-flags "-cp $out/lib/gol-tool.jar" \
```

Comment:
Unfamiliar with the -cp flag in the java command. Does this do the same thing as `-jar`?

Reply:
Almost the same according to https://stackoverflow.com/a/11922724
I used it because it was used in https://github.com/clarisma/gol-tool/blob/main/bin/gol